lib/ansible/modules/windows/win_xml.ps1

```diff
@@ -0,0 +1,182 @@
+#!powershell
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+Set-StrictMode -Version 2
+
+function Compare-XmlDocs($actual, $expected) {
+    if ($actual.Name -ne $expected.Name) {
+        throw "Actual name not same as expected: actual=" + $actual.Name + ", expected=" + $expected.Name
+    }
+    ##attributes...
+
+    if ($actual.Attributes -ne $null -and $expected.Attributes -ne $null) {
+        if ($actual.Attributes.Count -ne $expected.Attributes.Count) {
+           throw "attribute mismatch for actual=" + $actual.Name
+        }
+        for ($i=0;$i -lt $expected.Attributes.Count; $i =$i+1) {
+            if ($expected.Attributes[$i].Name -ne $actual.Attributes[$i].Name) {
+                throw "attribute name mismatch for actual=" + $actual.Name
+            }
+            if ($expected.Attributes[$i].Value -ne $actual.Attributes[$i].Value) {
+                throw "attribute value mismatch for actual=" + $actual.Name
+            }
+        }
+    }
```

Comment:
Lines 29-31 check the attribute counts so that covers the check as specified. I'm pretty sure the logic in lines 32-38 covers the checks to make sure that the child node is not there. The issue might be complex if a similar child is there, then what is the intent?